### PR TITLE
Several fixes, consistency changes, and optimizations

### PR DESCRIPTION
**Pull request by krischerven**
- make the program not panic on bad search queries (which breaks some terminals)
- make the program fail to run when given invalid arguments
- add podcast-cli and Vim files to .gitignore
- consistency changes ([]*Episode{} -> make([]*Episode, 0), etc)
- slice pre-allocation optimizations
- formatting changes

### CODE REVIEW

File: podcasts/episodes.go [hunk 1]
```diff
@@ -39,22 +39,23 @@ func (podcast *Podcast) Episodes(index int, count int) []*Episode {
 	for i := 0; i < pagesToFetch; i++ {
 		go func(index int) {
 			fetchedEpisodes, err := podcast.scrapEpisodesByPodcast(fmt.Sprintf("%s/-episodes?page=%d", podcast.URL, startPage+index))
-			if err != nil {
+			if err == nil {
+				episodesChan <- fetchedEpisodes
+			} else {
 				log.Println(err)
 			}
```
Comment (goulinkh) [inline podcasts/episodes.go]: the problem is if there is an error there will be a deadlock at line 52

File: podcasts/episodes.go [hunk 2]
```diff
@@ -146,6 +147,6 @@ func SearchEpisode(episodes []*Episode, query string) []*Episode {
 
 type ByDate []*Episode
 
-func (a ByDate) Len() int           { return len(a) }
-func (a ByDate) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
-func (a ByDate) Less(i, j int) bool { return a[i].ReleaseDate.After(a[j].ReleaseDate) }
+func (b ByDate) Len() int           { return len(b) }
+func (b ByDate) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b ByDate) Less(i, j int) bool { return b[i].ReleaseDate.After(b[j].ReleaseDate) }
```
Comment (goulinkh) [inline podcasts/episodes.go:151]: I don't understand why you changed the variable name

**Review by goulinkh — CHANGES_REQUESTED** [review]

File: podcasts/episodes.go (hunk 1 above)
Comment (krischerven) [reply podcasts/episodes.go]: Didn't realize that. Will revert this change.

File: podcasts/episodes.go (hunk 2 above)
Comment (krischerven) [reply podcasts/episodes.go:151]: (b ByDate) seems more consistent with other naming conventions in the code (e *Episode)

**Review by goulinkh — APPROVED** [review]
@krischerven Awesome work thank you for your time

**Comment by krischerven** [comment]
No problem

**Pull request merged**